### ISSUE TITLE
Lookup parameter is updated after set_updated(False)

### ISSUE THREAD

**Issue by danielhers** [clab/dynet#1202]
Calling `set_updated(False)` on a parameter should cause that parameter not to be updated, but it seems to be ignored. Am I using it incorrectly?

Minimal working example:
```
import dynet_config
dynet_config.set(weight_decay=0)
import dynet as dy
import numpy as np
from numpy.testing import assert_array_equal

m = dy.ParameterCollection()
p = m.add_lookup_parameters((2, 2))
p.set_updated(False)
init = np.array(((2, 3), (4, 5)), dtype=float)
p.init_from_array(init)
t = dy.SimpleSGDTrainer(m)
e = dy.sum_elems(p.expr())
e.forward()
e.backward()
t.update()
dy.renew_cg()
assert_array_equal(p.as_array(), init)
# >>> AssertionError: 
# Arrays are not equal
#
# (mismatch 100.0%)
#  x: array([[1.9, 2.9],
#        [3.9, 4.9]])
#  y: array([[2., 3.],
#        [4., 5.]])
```
Calling `expr(update=False)` fixes the problem, but I think `set_updated` should be enough.

**Comment by xunzhang** [comment]
I just took a look, it's a bug in the current implementation. I fixed it in above pull request, please review. The `expr(update=False)` works since it will add a const parameter (const parameters will not be added in the `parameter_nodes` variable of a cg).

**Comment by neubig** [comment]
Fixed by #1203

**Comment by danielhers** [comment]
But it hasn't been merged yet, has it?
